### ISSUE TITLE
30% reduction of message index CSS (you know you want it)

### ISSUE THREAD

**Issue by StealthWombat** [SimpleMachines/SMF#3043]
I'll just leave this here and you can implement it when you feel like it. Went through the message index section of index.css and trimmed it substantially without changing the look. Also improved RTL support and slightly reduced markup at the same time. Code changes are simple.

So old CSS was this:

```
/* MessageIndex */
/* Topic list */
#topic_container .windowbg, #topic_container .windowbg2 {
height: 4em;
border: 1px solid #ddd;
border-top: none;
}
#topic_header .board_icon, #topic_header .info, #topic_header .board_stats, #topic_header .lastpost, #topic_header .moderation {
height: 2em;
font-weight: bold;
font-size: 1em;
margin-top: 3px;
margin-bottom: 0;
padding: 0;
}
#topic_header.title_bar {
margin: 0px;
border-radius: 0;
}
#messageindex .board_stats, #recent .board_stats {
text-align: center;
}
#topic_header input {
margin-top: 5px !important;
}
.moderation {
float: right;
width: 35px;
margin: 5px 0 0 0;
}
#topic_container .board_icon, #topic_container .info, #topic_container .windowbg2, #topic_container .windowbg, #topic_container .board_stats, #topic_container .lastpost {
box-shadow: none;
border-radius: 0;
padding: 0;
margin: 0;
overflow: hidden;
height: 3.5em;
}
#topic_container .icons span {
margin: 2px 0 0 5px;
}
.icon img, .moderation input {
margin-top: 15px;
}
.info .icons, #messageindex .board_stats p, .lastpost p, #unread .board_stats p, #unreadreplies .board_stats p {
margin-top: 5px;
border: none;
}
/* Start with description and other things */
#description_board {
padding: 8px 10px 8px 10px;
border-bottom-left-radius: 0;
border-bottom-right-radius: 0;
border-bottom: 0;
box-shadow: none;
}
.lastpost span.last_post {
float: right;
margin: 4px;
}
select.qaction {
padding: 0;
}
#quick_actions {
padding-right: 4px;
margin-top: 5px;
}
.qaction_cell .button_submit {
margin: 2px 4px 0 4px;
}
#topic_icons {
margin-top: 10px;
float: left;
width: 100%;
}
#topic_icons .information {
border-radius: 7px;
}
#topic_icons p {
line-height: 2em;
}
#topic_icons p.floatleft {
padding: 0 12px 2px 0;
}
/* Just to correct Fred there without requiring an extra image. Should make a new Fred with clear padding. */
#topic_icons p.floatleft img:first-child {
padding: 0 2px;
}
#message_index_jump_to_select {
font-size: 1em;
padding: 2px;
}
#message_index_jump_to .button_submit {
padding: 4px 6px 2px 6px;
}
```

You can replace that with this for extra coolness:

```
/* MessageIndex */
/* Start with description and other things */
#description_board {
    padding: 8px 10px;
    border-radius: 6px 6px 0 0;
    border-bottom: none;
    box-shadow: none;
}
#description_board h3, #description_board p {
    display: inline-block;
}
#description_board h3:after {
    content: ' - ';
}
/* Topic list */
#topic_header {
    margin: 0;
}
#topic_header div {
    height: 2em;
    font-weight: bold;
    padding: 0;
}
.moderation {
    display: table-cell;
    width: 35px;
    vertical-align: middle;
}
#topic_container .windowbg, #topic_container .windowbg2 {
    height: 4em;
    border: 1px solid #ddd;
    border-top: none;
}
#topic_container .windowbg, #topic_container .windowbg2 {
    box-shadow: none;
    border-radius: 0;
    padding: 0;
    margin: 0;
}
#topic_container .windowbg>div, #topic_container .windowbg2>div {
    padding-top: 0;
}
#topic_container .board_stats p {
    margin: .5em 0 -.5em;
}
/* Quick moderation selects and submit */
#quick_actions {
    padding: 4px 0;
}
.qaction {
    padding: 0;
}
/* Investigate default select and button sizing. Something is slightly off there.*/
.qaction.button_submit {
    margin: 1px 4px 0 4px;
}
/* Icons and jump to */
#topic_icons:before {
    display: block;
    height: 8px;
    clear: both;
    content: '';
}
#topic_icons .information {
    font-size: .9em;
    border-radius: 4px;
}
#topic_icons p {
    padding: 0 12px 0 4px;
    line-height: 2em;
}
/* Should lose this before RC1. It's a kludge. Can be fixed by new image */
#topic_icons .floatleft img:first-child {
    padding: 0 2px;
}
#message_index_jump_to_select {
    font-size: 1em;
    padding: 0;
}
/* This is a bit of a kludge too */
#message_index_jump_to .button_submit {
    margin-top: 3px;
    padding: 1px 6px;
}
```

When testing this for RTL I found that there's a little trap later in the file too. The .title_bar class has a text-align: left; declaration, which does nothing except screw RTL. That should be removed. Change to:

```
/* TitleBar & SubBar */
.title_bar {
    border-right: 1px solid #ddd;
    border-left: 1px solid #ddd;
    border-top: 2px solid #ff9400;
    border-bottom: 2px solid #bf6900;
    background: #fff;
    color: #666;
    border-radius: 2px 2px 0 0;
    margin: 5px 0 0 0;
}
```

Markup changes are really basic. I just ditched the floatleft on the description_board h3 and set it and the paragraph to inline-block for auto RTL. Also moved the hyphen and spaces to a pseudo in case any themers want to change those easily.

```
    <div id="description_board" class="generic_list_wrapper">
        <h3>', $context['name'], '</h3>
```

Then I removed the floatright class from the five generic_icons spans (watch, lock, sticky, move, poll) and put it on their parent .icons div to save some markup and CSS.

MessageIndex.template.php

```
            // Now we handle the icons
            echo '
                            <div class="icons floatright">';
            if ($topic['is_watched'])
                echo '
                                <span class="generic_icons watch" title="', $txt['watching_this_topic'], '"></span>';
            if ($topic['is_locked'])
                echo '
                                <span class="generic_icons lock"></span>';
            if ($topic['is_sticky'])
                echo '
                                <span class="generic_icons sticky"></span>';
            if ($topic['is_redirect'])
                echo '
                                <span class="generic_icons move"></span>';
            if ($topic['is_poll'])
                echo '
                                <span class="generic_icons poll"></span>';
            echo '
                            </div>';
```

Also needs that tweak in Recent.template.php. Exactly the same change in both function template_unread() and function template_replies()

```
            // Now we handle the icons
            echo '
                                <div class="icons floatright">';
            if ($topic['is_locked'])
                echo '
                                    <span class="generic_icons lock"></span>';
            if ($topic['is_sticky'])
                echo '
                                    <span class="generic_icons sticky"></span>';
            if ($topic['is_poll'])
                echo '
                                    <span class="generic_icons poll"></span>';
            echo '
                                </div>';
```

Merry Christmas n stuff.

ETA: Meh, GitHub makes an awful mess of code. :P I'll post it in code tags on SMF so it can just be copy/pasted without having to reformat it.

**Comment by StealthWombat** [comment]
Ok, formatted code here if anyone wants it: http://www.simplemachines.org/community/index.php?topic=539574.msg3835855#msg3835855

**Comment by Oldiesmann** [comment]
The code is a mess because you need three ` marks, not two...

```
/* MessageIndex */
/* Topic list */
#topic_container .windowbg, #topic_container .windowbg2 {
height: 4em;
border: 1px solid #ddd;
border-top: none;
}
#topic_header .board_icon, #topic_header .info, #topic_header .board_stats, #topic_header .lastpost, #topic_header .moderation {
height: 2em;
font-weight: bold;
font-size: 1em;
margin-top: 3px;
margin-bottom: 0;
padding: 0;
}
#topic_header.title_bar {
margin: 0px;
border-radius: 0;
}
#messageindex .board_stats, #recent .board_stats {
text-align: center;
}
#topic_header input {
margin-top: 5px !important;
}
.moderation {
float: right;
width: 35px;
margin: 5px 0 0 0;
}
#topic_container .board_icon, #topic_container .info, #topic_container .windowbg2, #topic_container .windowbg, #topic_container .board_stats, #topic_container .lastpost {
box-shadow: none;
border-radius: 0;
padding: 0;
margin: 0;
overflow: hidden;
height: 3.5em;
}
#topic_container .icons span {
margin: 2px 0 0 5px;
}
.icon img, .moderation input {
margin-top: 15px;
}
.info .icons, #messageindex .board_stats p, .lastpost p, #unread .board_stats p, #unreadreplies .board_stats p {
margin-top: 5px;
border: none;
}
/* Start with description and other things */
#description_board {
padding: 8px 10px 8px 10px;
border-bottom-left-radius: 0;
border-bottom-right-radius: 0;
border-bottom: 0;
box-shadow: none;
}
.lastpost span.last_post {
float: right;
margin: 4px;
}
select.qaction {
padding: 0;
}
#quick_actions {
padding-right: 4px;
margin-top: 5px;
}
.qaction_cell .button_submit {
margin: 2px 4px 0 4px;
}
#topic_icons {
margin-top: 10px;
float: left;
width: 100%;
}
#topic_icons .information {
border-radius: 7px;
}
#topic_icons p {
line-height: 2em;
}
#topic_icons p.floatleft {
padding: 0 12px 2px 0;
}
/* Just to correct Fred there without requiring an extra image. Should make a new Fred with clear padding. */
#topic_icons p.floatleft img:first-child {
padding: 0 2px;
}
#message_index_jump_to_select {
font-size: 1em;
padding: 2px;
}
#message_index_jump_to .button_submit {
padding: 4px 6px 2px 6px;
}
``

You can replace that with this for extra coolness:

``
/* MessageIndex */
/* Start with description and other things */
#description_board {
    padding: 8px 10px;
    border-radius: 6px 6px 0 0;
    border-bottom: none;
    box-shadow: none;
}
#description_board h3, #description_board p {
    display: inline-block;
}
#description_board h3:after {
    content: ' - ';
}
/* Topic list */
#topic_header {
    margin: 0;
}
#topic_header div {
    height: 2em;
    font-weight: bold;
    padding: 0;
}
.moderation {
    display: table-cell;
    width: 35px;
    vertical-align: middle;
}
#topic_container .windowbg, #topic_container .windowbg2 {
    height: 4em;
    border: 1px solid #ddd;
    border-top: none;
}
#topic_container .windowbg, #topic_container .windowbg2 {
    box-shadow: none;
    border-radius: 0;
    padding: 0;
    margin: 0;
}
#topic_container .windowbg>div, #topic_container .windowbg2>div {
    padding-top: 0;
}
#topic_container .board_stats p {
    margin: .5em 0 -.5em;
}
/* Quick moderation selects and submit */
#quick_actions {
    padding: 4px 0;
}
.qaction {
    padding: 0;
}
/* Investigate default select and button sizing. Something is slightly off there.*/
.qaction.button_submit {
    margin: 1px 4px 0 4px;
}
/* Icons and jump to */
#topic_icons:before {
    display: block;
    height: 8px;
    clear: both;
    content: '';
}
#topic_icons .information {
    font-size: .9em;
    border-radius: 4px;
}
#topic_icons p {
    padding: 0 12px 0 4px;
    line-height: 2em;
}
/* Should lose this before RC1. It's a kludge. Can be fixed by new image */
#topic_icons .floatleft img:first-child {
    padding: 0 2px;
}
#message_index_jump_to_select {
    font-size: 1em;
    padding: 0;
}
/* This is a bit of a kludge too */
#message_index_jump_to .button_submit {
    margin-top: 3px;
    padding: 1px 6px;
}
```

**Comment by StealthWombat** [comment]
Yeah Antes put me onto that. I edited my OP. Three is pretty silly. Let's make it ten or twelve. :P

**Comment by XinYenFon** [comment]
https://github.com/Antes/kittenground/commit/4ee5d6c0cd0c70ab4b18a6b45886cee24f297212 - Are we good here? Your code had some duplicates, merged & removed some (very little to none).

**Comment by StealthWombat** [comment]
Antes: not really. For a start, good catch on the duplicate declaration for #topic_container .windowbg, #topic_container .windowbg2. That should indeed be merged into:

```
#topic_container .windowbg, #topic_container .windowbg2 {
    height: 4em;
    margin: 0;
    padding: 0;
    border: 1px solid #ddd;
    border-top: none;
    border-radius: 0;
    box-shadow: none;
}
```

The markup looks fine, however I am reluctant to agree to the rest of the CSS.

The first problem, as I see it, is that I deliberately re-ordered the CSS so it followed the order of the markup. Doing it this way makes CSS files much easier to follow. If someone is looking for the first items in the message index (h3 and description) they'll expect to see them at the top of the message index section. Having them halfway down, as in the old version, doesn't make sense.

The second problem is that by trying to merge the new stuff into the old stuff, you've still ended up with excessively verbose declarations and/or redundant declarations that only bloat the file for no benefit. I got rid of all of those on purpose.

 It really would be better to just do a complete find/replace on the entire message index section of the CSS. It would also be less work. :D

I've tested it in LTR and RTL and as far as I can tell it's fine. It may need some tweaking for some responsive breakpoints, but a quick check there didn't show anything immediately obvious.

If you're nervous about just replacing the entire block, just leave the commit until you have time to try it on local.

ETA: You have also missed the removal of text-align: left; from .title_bar later in the file, which as I mentioned does nothing to help LTR and only borks RTL. :)

Is it possible to attach files here? I could just give you my index.css from local if it's useful.

**Comment by XinYenFon** [comment]
I'm not sure, you can always attach to SMorg. I did find/replace actually then inspected some elements and find duplicates.

```
/* MessageIndex */
/* Topic list */
#topic_header .board_icon, #topic_header .info, #topic_header .board_stats, #topic_header .lastpost, #topic_header .moderation {
    height: 2em;
    font-weight: bold;
    font-size: 1em;
    margin-top: 3px;
    margin-bottom: 0;
    padding: 0;
}
#topic_header.title_bar {
    margin: 0px;
    border-radius: 0;
}
#messageindex .board_stats, #recent .board_stats {
    text-align: center;
}
#topic_header input {
    margin-top: 5px !important;
}
.moderation {
    float: right;
    width: 35px;
    margin: 5px 0 0 0;
}
#topic_container .board_icon, #topic_container .info, #topic_container .windowbg2, #topic_container .windowbg, #topic_container .board_stats, #topic_container .lastpost {
    box-shadow: none;
    border-radius: 0;
    padding: 0;
    margin: 0;
    overflow: hidden;
    height: 4em;
}
#topic_container .windowbg, #topic_container .windowbg2 {
    border: 1px solid #ddd;
    border-top: none;
}
#topic_container .icons span {
    margin: 2px 0 0 5px;
}
.icon img, .moderation input {
    margin-top: 15px;
}
.info .icons, #messageindex .board_stats p, .lastpost p, #unread .board_stats p, #unreadreplies .board_stats p {
    margin-top: 5px;
    border: none;
}
/* MessageIndex */
/* Start with description and other things */
#description_board {
    padding: 8px 10px;
    border-radius: 6px 6px 0 0;
    border-bottom: none;
    box-shadow: none;
}
#description_board h3, #description_board p {
    display: inline-block;
}
#description_board h3:after {
    content: ' - ';
}
/* Topic list */
.moderation {
    display: table-cell;
    width: 35px;
    vertical-align: middle;
}

/* Quick moderation selects and submit */
#quick_actions {
    padding: 4px 0;
}
.qaction {
    padding: 0;
}
/* Investigate default select and button sizing. Something is slightly off there.*/
.qaction.button_submit {
    margin: 1px 4px 0 4px;
}
/* Icons and jump to */
#topic_icons:before {
    display: block;
    height: 8px;
    clear: both;
    content: '';
}
#topic_icons .information {
    font-size: .9em;
    border-radius: 4px;
}
#topic_icons p {
    padding: 0 12px 0 4px;
    line-height: 2em;
}
/* Should lose this before RC1. It's a kludge. Can be fixed by new image */
#topic_icons .floatleft img:first-child {
    padding: 0 2px;
}
#message_index_jump_to_select {
    font-size: 1em;
    padding: 0;
}
/* This is a bit of a kludge too */
#message_index_jump_to .button_submit {
    margin-top: 3px;
    padding: 0px 6px;
}
```

**Comment by StealthWombat** [comment]
But you're still declaring piles of stuff which wasn't in the code I provided.

For example, the code I gave doesn't have this anywhere at all:

```
#topic_header .board_icon, #topic_header .info, #topic_header .board_stats, #topic_header .lastpost, #topic_header .moderation {
    height: 2em;
    font-weight: bold;
    font-size: 1em;
    margin-top: 3px;
    margin-bottom: 0;
    padding: 0;
}
```

Instead, I used the much shorter but equally effective:

```
#topic_header div {
    height: 2em;
    font-weight: bold;
    padding: 0;
}
```

Did you copy from me, or did you copy from Oldies? I think Oldies just pasted everything: old code and new code all together in one go. If you copied that by mistake then sure, it'd make an awful mess.

**Comment by StealthWombat** [comment]
Ok, here's the code I was recommending again, without any Oldies additions. :D

This is just the replace stuff for the whole message index section. I've double checked it for duplicates and fixed the only one.

```
/* MessageIndex */
/* Start with description and other things */
#description_board {
    padding: 8px 10px;
    border-radius: 6px 6px 0 0;
    border-bottom: none;
    box-shadow: none;
}
#description_board h3, #description_board p {
    display: inline-block;
}
#description_board h3:after {
    content: ' - ';
}
/* Topic list */
#topic_header {
    margin: 0;
}
#topic_header div {
    height: 2em;
    font-weight: bold;
    padding: 0;
}
.moderation {
    display: table-cell;
    width: 35px;
    vertical-align: middle;
}
#topic_container .windowbg, #topic_container .windowbg2 {
    height: 4em;
    margin: 0;
    padding: 0;
    border: 1px solid #ddd;
    border-top: none;
    border-radius: 0;
    box-shadow: none;
}
#topic_container .windowbg>div, #topic_container .windowbg2>div {
    padding-top: 0;
}
#topic_container .board_stats p {
    margin: .5em 0 -.5em;
}
/* Quick moderation selects and submit */
#quick_actions {
    padding: 4px 0;
}
.qaction {
    padding: 0;
}
/* Investigate default select and button sizing. Something is slightly off there.*/
.qaction.button_submit {
    margin: 1px 4px 0 4px;
}
/* Icons and jump to */
#topic_icons:before {
    display: block;
    height: 8px;
    clear: both;
    content: '';
}
#topic_icons .information {
    font-size: .9em;
    border-radius: 4px;
}
#topic_icons p {
    padding: 0 12px 0 4px;
    line-height: 2em;
}
/* Should lose this before RC1. It's a kludge. Can be fixed by new image */
#topic_icons .floatleft img:first-child {
    padding: 0 2px;
}
#message_index_jump_to_select {
    font-size: 1em;
    padding: 0;
}
/* This is a bit of a kludge too */
#message_index_jump_to .button_submit {
    margin-top: 3px;
    padding: 1px 6px;
}
```

Plus there's this small change later in the file, which just removes the text-align from .title_bar:

```
/* TitleBar & SubBar */
.title_bar {
    border-right: 1px solid #ddd;
    border-left: 1px solid #ddd;
    border-top: 2px solid #ff9400;
    border-bottom: 2px solid #bf6900;
    background: #fff;
    color: #666;
    border-radius: 2px 2px 0 0;
    margin: 5px 0 0 0;
}
```

**Comment by StealthWombat** [comment]
That commit still has a lot of crud in it. I'm giving up for the night. Later.

**Comment by XinYenFon** [comment]
@StealthWombat ??? That's your code... Copied from you

**Comment by StealthWombat** [comment]
Oh yeah, it's showing up now. Cool. When I tried it last night it was still showing old code. Does GitHub take a while to update the file view sometimes?

Because last night I went to https://github.com/Antes/kittenground/commit/9956299e7f535f9fe36138f6a2136a2ed1fa0321 then hit the View button, and it was still showing the code from your post back here: https://github.com/SimpleMachines/SMF2.1/issues/3043#issuecomment-139165447

Anyway it's sorted so that's the main thing. Thanks.

**Comment by StealthWombat** [comment]
Is this supposed to be in the 2.1 release repo yet?

(Just cloned that and it's showing different code)

**Comment by XinYenFon** [comment]
Yeap. You don't have to clone the repo again and again, use git fetch upstream (define upstream first) then do git pull upstream release-2.1

https://gist.github.com/CristinaSolana/1885435

**Comment by StealthWombat** [comment]
Sorry, I meant I forked it, not cloned it. I cloned to desktop after forking it :)

Ok, this Github thing is weird. How is it that I fork the repo an hour or so ago, and it doesn't contain the new code? I don't get it.

And sorry, but that gist page means nothing to me.

**Comment by StealthWombat** [comment]
Ok, so I now have a synched local repo that has pulled down the latest from this repo. I can see exactly what's going on in the file now. There has been a balls up.

What has happened is that the code I gave has been added in alright, but it has been added in after some of the code that was there before.

Line 2947 of index.css is where the old code starts. Line 2995 is where the new code starts.

https://github.com/StealthWombat/SMF-totally-forked/issues/3

Result is that rather than a 30% reduction in crud, we haz 30% or thereabouts increase in crud.

_looks for dancing bananas emoticon_
_cannot find_
_haz a sad_

Anyway, silly buggers aside, what needs to happen now is to double check stuff before removing the remnants of the old code. Bit late for me to try that tonight, but will have another look tomorrow. Need my beauty sleep.
